### ISSUE TITLE
more dynamic detection of stuff in config.mk

### ISSUE THREAD

**Issue by ipelupessy** [amusecode/amuse#577]
**Is your feature request related to a problem? Please describe.**
config.mk is generated once at install time. Most of the stuff in there is only needed later at compile.

**Describe the solution you'd like**
a mechanism whereby detection proceeds at compile time (or maybe a possibility of regenerating it w/o reinstall)

**Describe alternatives you've considered**
telling everyone to reinstall if the environment is updated

**Additional context**
clearly this needs some thought...

**Comment by rieder** [comment]
Yeah that sounds like the right mechanism. The alternative mentioned is what we have now and it isn't great.
Another alternative could be having individual config.mk files for each package.

**Comment by ipelupessy** [comment]
see #583
